auth/postgres/repo.go

```diff
@@ -63,18 +63,34 @@ func (pr *patRepo) Retrieve(ctx context.Context, userID, patID string) (auth.PAT
 }
 
 func (pr *patRepo) RetrieveAll(ctx context.Context, userID string, pm auth.PATSPageMeta) (auth.PATSPage, error) {
-	q := `
-		SELECT 
-		p.id, p.user_id, p.name, p.description, p.issued_at, p.expires_at,
-		p.updated_at, p.revoked, p.revoked_at
-		FROM pats p WHERE user_id = :user_id
-		ORDER BY issued_at DESC
-		LIMIT :limit OFFSET :offset`
+	pageQuery, err := PageQuery(pm)
+	if err != nil {
+		return auth.PATSPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+	}
+
+	q := fmt.Sprintf(`
+	WITH updated_pats AS (
+		UPDATE pats
+		SET status = %d
+		WHERE user_id = :user_id AND expires_at < :expires_at
+		RETURNING id
+	)
```

Comment:
I appreciate this query, but it looks like a hack. 
We are doing update while retrieving